spec/features/hyrax_doi_spec.rb

```diff
@@ -130,60 +130,29 @@ def work_url(work)
 
     context "when the user selects `findable`" do
       let(:new_title) { "New work title" }
-      let(:doi) { "#{prefix}/kgkc-nn31" }
-
-      let(:response_fixture) { File.read Rails.root.join("..", "fixtures", "datacite", "put_metadata.xml") }
-      let(:common_headers) do
-        {
-          "Accept": "*/*",
-          "Accept-Encoding": "gzip;q=1.0,deflate;q=0.6,identity;q=0.3",
-          "Authorization": "Basic VkpLQS5KQ1JYWkctTE9DQUw6cGFzc3dvcmQ=",
-          "User-Agent": "Faraday v0.17.4"
-        }
-      end
-      let(:json_headers) do
-        common_headers.merge("Content-Type": "application/vnd.api+json")
-      end
-      let(:xml_headers) do
-        common_headers.merge("Content-Type": "application/xml;charset=UTF-8")
-      end
-      let(:text_headers) do
-        common_headers.merge("Content-Type": "text/plain;charset=UTF-8")
+
+      it "redirects the user to the works page" do
+        fill_in_form(new_title)
+
+        expect(page).to have_selector("h1", text: "Work", wait: 10)
+        expect(page).to have_selector("h2", text: new_title)
       end
 
-      before do
-        # The initial request to create_draft_doi
-        stub_request(:post, "https://api.test.datacite.org/dois")
-          .with(body: { "data": { "type": "dois", "attributes": { "prefix": prefix } } }.to_json, headers: json_headers)
-          .to_return(status: 201, body: { "data": { "id": doi, "type": "dois", "attributes": {} } }.to_json, headers: {})
-
-        # Send the work data to datacite
-        stub_request(:put, "https://mds.test.datacite.org/metadata/#{doi}")
-          .with(body: response_fixture, headers: xml_headers)
-          .to_return(status: 201, body: "", headers: {})
-
-        # Register the URL
-        stub_request(:put, "https://mds.test.datacite.org/doi/#{doi}")
-          .with(body: "doi=#{doi}\nurl=http://#{cname}/concern/generic_works/#{work.id}", headers: text_headers)
-          .to_return(status: 201, body: "", headers: {})
+      it "enqueues a job to mint the DOI" do
+        expect { fill_in_form(new_title) }.to have_enqueued_job(Hyrax::DOI::RegisterDOIJob).with(work.reload, registrar: "datacite", registrar_opts: {})
```

Comment:
I think from the UI, just filling a title should not cause a doi to be minted. I think it is checkbox you need to check.

See attach image. So we need to check draft
![doi](https://user-images.githubusercontent.com/329514/144077819-a0251022-c225-43d1-a97c-7455b70fc93e.jpg)